packages/signer/signer-solana/src/internal/app-binder/device-action/SignMessage/SignMessageDeviceAction.ts

```
@@ -0,0 +1,241 @@
+              sendingData: new TextEncoder().encode(context.input.message),
+            }),
+            onDone: {
+              target: "SignMessageResultCheck",
+              actions: [
+                assign({
+                  _internalState: ({ event, context }) => {
+                    if (isSuccessCommandResult(event.output)) {
+                      return {
+                        ...context._internalState,
+                        signature: event.output.data,
+                      };
+                    }
+                    return {
+                      ...context._internalState,
+                      error: event.output.error,
+                    };
+                  },
+                }),
+              ],
+            },
+            onError: {
+              target: "Error",
+              actions: "assignErrorFromEvent",
+            },
+          },
+        },
+        SignMessageResultCheck: {
+          always: [
+            { guard: "noInternalError", target: "Success" },
+            { target: "Error" },
+          ],
+        },
+        Success: {
+          type: "final",
+        },
+        Error: {
+          type: "final",
+        },
+      },
+      output: ({ context }) =>
+        context._internalState.signature
+          ? Right(context._internalState.signature)
+          : Left(
+              context._internalState.error ||
+                new UnknownDAError("No error in final state"),
+            ),
+    });
+  }
+
+  extractDependencies(internalApi: InternalApi): MachineDependencies {
+    const signMessage = async (arg0: { input: SendSignMessageTaskArgs }) => {
+      const result = await new SendSignMessageTask(
+        internalApi,
+        arg0.input,
+      ).run();
+      if (isSuccessCommandResult(result)) {
+        if (result.data.isJust()) {
+          return {
+            ...result,
+            data: result.data.extract(),
+          };
+        } else {
```

[SHOULD] Now we don't have to split a message in chunk, we should get a signature on the command response. If we don't have any signature we should return an error.

I think this part is not necessary because we need to adapt the command to not return a `Maybe<Signature>` but a `Signature`.